### PR TITLE
feat: support Acts-20.2.0 and beyond (within 20.x release series)

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
In order to test the EPIC geometry upgrade to use Acts-20 construction, we need to have a reconstruction framework that is able to work with that Acts version. This upgrades EICrecon tracking algorithms to work with Acts-20.2.0 and beyond.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators: @DraTeots 

### Does this PR introduce breaking changes? What changes might users need to make to their code?
Yes, there is no backwards compatibility for older Acts versions.

### Does this PR change default behavior?
No.

### CODE REVIEW

**Comment by wdconinc** [comment]
DO NOT MERGE since this will require coordination with containers etc. Until then the builds will fail here since Acts-20 is not available in the container yet.

**Comment by rahmans1** [comment]
Ran 2000 events with the grandteton and brycecanyon configuration with ![epic refactored geometry].(https://github.com/eic/epic/pull/283)

```
ddsim --compactFile epic_${DETECTOR_CONFIG}.xml -G --gun.particle "pi-" --gun.momentumMin "1*GeV" --gun.momentumMax "20*GeV" --gun.distribution "uniform" -N 2000 --outputFile sim_pi-_1GeV_20GeV_${DETECTOR_CONFIG}.edm4hep.root

run_eicrecon_reco_flags.py sim_pi-_1GeV_20GeV_${DETECTOR_CONFIG}.edm4hep.root rec_pi-_1GeV_20GeV_${DETECTOR_CONFIG} -Pplugins=dump_flags,janadot -Pdump_flags:json=${DETECTOR_CONFIG}_right_flags.json
```

===Grandteton===
GrandtetonDistribution
===Brycecanyon===
BrycecanyonDistribution
===Grandteton===
GrandtetonLinear
===Brycecanyon===
BrycecanyonLinear

**Comment by wdconinc** [comment]
These look good. Next steps are merging acts into container and merging this into main. I've marked as ready for review.

**Comment by wdconinc** [comment]
@faustus123 This is a PR that we would like to merge tomorrow, so we can settle down the geometry and containers for running production with Acts-20 over the break.

File: cmake/jana_plugin.cmake [hunk 1]
```diff
@@ -176,7 +176,7 @@ macro(plugin_add_acts _name)
 
     if(NOT Acts_FOUND)
         find_package(Acts REQUIRED COMPONENTS Core PluginIdentification PluginTGeo PluginJson PluginDD4hep)
-        set(Acts_VERSION_MIN "19.0.0")
+        set(Acts_VERSION_MIN "20.2.0")
```
Comment (sly2j) [inline cmake/jana_plugin.cmake:179]: 20.2 (versus 20.0) is a hard requirement?

Comment (wdconinc) [reply cmake/jana_plugin.cmake:179]: Yeah, acts-20.2.0 introduced this change:
```diff
  std::optional<MultiTrajectory> m_multiTrajectory;
  std::shared_ptr<MultiTrajectory> m_multiTrajectory;
```
The way they get around this in a minor release is that technically this is a change in their examples, not their main interfaces.

Comment (wdconinc) [reply cmake/jana_plugin.cmake:179]: Ref: https://github.com/acts-project/acts/pull/1507

**Review by sly2j — APPROVED** [review]

**Pull request merged**